include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -191,13 +221,10 @@ namespace seqan3
  * \attention This is a concept requirement, not an actual function (however types
  *            modelling this concept will provide an implementation).
  */
-//!\}
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT aligned_sequence =
-    sequence<t> &&
-    std::ranges::forward_range<t> &&
-    weakly_assignable_from<std::ranges::range_reference_t<t>, gap const &> &&
+SEQAN3_CONCEPT writable_aligned_sequence =
+    aligned_sequence<t> &&
     requires { typename detail::unaligned_seq_t<t>; } &&
     requires (t v, detail::unaligned_seq_t<t> unaligned)
     {
```

Comment:
Yeah I also wondered why we had this in the first place because we have `insert_gap` and I don't even know if the gap decorator iterator is an output iterator.

Reply:
Hmm, I know that `gap_decorator` models `old::aligned_sequence`, there must be function overload for this. Ah I see gapped decorator returns a "proxy" `gapped<alphabet> &&` which fulfils this. We lose output range with this, we might be able to get it back if we would return a `gap_decorator::proxy<alphabet>` which returns a conditional modifiable reference to the underlying range and is equally comparable to `seqan3::gap`. Conditional, because if it is a gap, we can't modify that gap.

Reply:
> We lose output range with this, we might be able to get it back if we would return a gap_decorator::proxy<alphabet> which returns a conditional modifiable reference to the underlying range and is equally comparable to seqan3::gap. Conditional, because if it is a gap, we can't modify that gap.

That would be a bad design choice to do something conditionally. If you want the gap decorator to model output_range, you need the underlying range to be writable as well. Basically that means something like `insert` is defined on that type. Then you can remove the gap and add the value at the corresponding position. And you know what. We can get this with a journal_decorator which allows to be a writable range without changing the underlying range.